spec/controllers/campus_access_exceptions_controller_spec.rb

```diff
@@ -24,6 +24,12 @@
         expect(assigns["campus_access_filename"]).to eq("#{ENV['CAMPUS_ACCESS_DIRECTORY']}/additional_campus_access.csv")
         expect(File.exist?(assigns["campus_access_filename"])).to be_truthy
       end
+
+      it "requires the file" do
+        post :create, params: { exception_file: nil }
+        expect(response).to redirect_to("http://test.host/campus_access_exceptions/new")
+        expect(flash[:alert]).to eq('param is missing or the value is empty: exception_file')
+      end
```

Comment:
I can't see the flash alert on the ui.

Reply:
@christinach you would need to send data directly to the controller to ever see this.  The form required keeps you from sending this data via the UI.
![Screen Shot 2021-01-06 at 1 48 31 PM](https://user-images.githubusercontent.com/1599081/103808374-e4594600-5025-11eb-88b1-f01416314f1e.png)

Reply:
yes I noticed the 'require=true'. but in the ui what action will you make to have the flash alert since the require: true does not allow you to submit no file?

Reply:
maybe I'm confusing you and my self. I dont know how the notification as a flash alert will be triggered since I can't submit an empty file because of the client restriction. required:true. So I dont know if the flash alert is needed.